Sorting by publication in admin interface API results in an internal server error

When sorting by publication in the admin interface event API like `/admin-ng/event/events.json?limit=100&offset=0&sort=publication:ASC` an internal server error will occur.

<details>
<summary>Full stack trace</summary>

```
2020-08-09T16:28:28,252 | ERROR | (AbstractEventEndpoint:2404) - The admin UI Search Index was not able to get the events list:
org.opencastproject.matterhorn.search.SearchIndexException: Error querying event index
	at org.opencastproject.index.service.impl.index.AbstractSearchIndex.getByQuery(AbstractSearchIndex.java:481) ~[!/:?]
	at org.opencastproject.adminui.endpoint.AbstractEventEndpoint.getEvents(AbstractEventEndpoint.java:2402) [!/:?]
	at sun.reflect.GeneratedMethodAccessor74.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_262]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_262]
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) [!/:3.3.3]
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) [!/:3.3.3]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201) [!/:3.3.3]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104) [!/:3.3.3]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) [!/:3.3.3]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) [!/:3.3.3]
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308) [!/:3.3.3]
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) [!/:3.3.3]
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:267) [!/:3.3.3]
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) [!/:3.3.3]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) [!/:3.3.3]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) [!/:3.3.3]
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:216) [!/:3.3.3]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:301) [!/:3.3.3]
	at org.opencastproject.kernel.rest.RestPublisher$RestServlet.handleRequest(RestPublisher.java:558) [!/:?]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doGet(AbstractHTTPServlet.java:225) [!/:3.3.3]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687) [!/:3.1.0]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:276) [!/:3.3.3]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:852) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604) [!/:9.4.20.v20190813]
	at org.opencastproject.kernel.rest.JsonpFilter.doFilter(JsonpFilter.java:113) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.opencastproject.kernel.filter.proxy.TransparentProxyFilter.doFilter(TransparentProxyFilter.java:63) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.opencastproject.security.urlsigning.filter.UrlSigningFilter.doFilter(UrlSigningFilter.java:86) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.opencastproject.kernel.security.RemoteUserAndOrganizationFilter.doFilter(RemoteUserAndOrganizationFilter.java:247) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.opencastproject.kernel.rest.CurrentJobFilter.doFilter(CurrentJobFilter.java:87) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:118) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:84) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.opencastproject.kernel.security.AsyncTimeoutRedirectFilter.doFilter(AsyncTimeoutRedirectFilter.java:60) [!/:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:113) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:103) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:113) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:146) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:54) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:45) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:98) [!/:3.1.4.RELEASE]
	at org.springframework.security.oauth.provider.filter.OAuthProviderProcessingFilter.doFilter(OAuthProviderProcessingFilter.java:182) [!/:?]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
	at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilter(BasicAuthenticationFilter.java:150) [!/:3.1.7.RELEASE]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
	at org.springframework.security.web.authentication.www.DigestAuthenticationFilter.doFilter(DigestAuthenticationFilter.java:115) [!/:3.1.7.RELEASE]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
	at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:82) [!/:3.1.4.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.ui.DefaultLoginPageGeneratingFilter.doFilter(DefaultLoginPageGeneratingFilter.java:91) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:183) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:105) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160) [!/:3.1.7.RELEASE]
	at org.opencastproject.kernel.security.SecurityFilter.doFilter(SecurityFilter.java:124) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.opencastproject.kernel.security.OrganizationFilter.doFilter(OrganizationFilter.java:135) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.opencastproject.kernel.filter.https.HttpsFilter.doFilter(HttpsFilter.java:73) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.opencastproject.kernel.rest.CleanSessionsFilter.doFilter(CleanSessionsFilter.java:89) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1583) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:226) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:542) [!/:9.4.20.v20190813]
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:71) [!/:?]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:536) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1581) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1307) [!/:9.4.20.v20190813]
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:293) [!/:?]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:482) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1549) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1204) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [!/:9.4.20.v20190813]
	at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) [!/:?]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.Server.handle(Server.java:494) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:374) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:367) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782) [!/:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918) [!/:9.4.20.v20190813]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_262]
Caused by: org.opencastproject.matterhorn.search.SearchIndexException: ElasticsearchStatusException[Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]]
	at org.opencastproject.index.service.impl.index.AbstractSearchIndex.executeQuery(AbstractSearchIndex.java:618) ~[!/:?]
	at org.opencastproject.index.service.impl.index.AbstractSearchIndex.getByQuery(AbstractSearchIndex.java:473) ~[!/:?]
	... 113 more
Caused by: org.elasticsearch.ElasticsearchStatusException: Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]
	at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:177) ~[!/:?]
	at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:1727) ~[!/:?]
	at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:1704) ~[!/:?]
	at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1467) ~[!/:?]
	at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1424) ~[!/:?]
	at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1394) ~[!/:?]
	at org.elasticsearch.client.RestHighLevelClient.search(RestHighLevelClient.java:930) ~[!/:?]
	at org.opencastproject.index.service.impl.index.AbstractSearchIndex.executeQuery(AbstractSearchIndex.java:616) ~[!/:?]
	at org.opencastproject.index.service.impl.index.AbstractSearchIndex.getByQuery(AbstractSearchIndex.java:473) ~[!/:?]
	... 113 more
	Suppressed: org.elasticsearch.client.ResponseException: method [POST], host [http://localhost:9200], URI [/adminui_event/_search?typed_keys=true&ignore_unavailable=false&expand_wildcards=open&allow_no_indices=true&preference=_local&ignore_throttled=true&search_type=query_then_fetch&batched_reduce_size=512&ccs_minimize_roundtrips=true], status line [HTTP/1.1 400 Bad Request]
{"error":{"root_cause":[{"type":"query_shard_exception","reason":"No mapping found for [publication] in order to sort on","index_uuid":"SNjee3S7TYC1MmAvfRLaDg","index":"adminui_event"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"adminui_event","node":"EwV8Iw4JTwqXfW-2uTtm-g","reason":{"type":"query_shard_exception","reason":"No mapping found for [publication] in order to sort on","index_uuid":"SNjee3S7TYC1MmAvfRLaDg","index":"adminui_event"}}]},"status":400}
		at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:253) ~[!/:?]
		at org.elasticsearch.client.RestClient.performRequest(RestClient.java:231) ~[!/:?]
		at org.elasticsearch.client.RestClient.performRequest(RestClient.java:205) ~[!/:?]
		at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1454) ~[!/:?]
		at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1424) ~[!/:?]
		at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1394) ~[!/:?]
		at org.elasticsearch.client.RestHighLevelClient.search(RestHighLevelClient.java:930) ~[!/:?]
		at org.opencastproject.index.service.impl.index.AbstractSearchIndex.executeQuery(AbstractSearchIndex.java:616) ~[!/:?]
		at org.opencastproject.index.service.impl.index.AbstractSearchIndex.getByQuery(AbstractSearchIndex.java:473) ~[!/:?]
		at org.opencastproject.adminui.endpoint.AbstractEventEndpoint.getEvents(AbstractEventEndpoint.java:2402) [!/:?]
		at sun.reflect.GeneratedMethodAccessor74.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_262]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_262]
		at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) [!/:3.3.3]
		at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) [!/:3.3.3]
		at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201) [!/:3.3.3]
		at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104) [!/:3.3.3]
		at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) [!/:3.3.3]
		at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) [!/:3.3.3]
		at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308) [!/:3.3.3]
		at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) [!/:3.3.3]
		at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:267) [!/:3.3.3]
		at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) [!/:3.3.3]
		at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) [!/:3.3.3]
		at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) [!/:3.3.3]
		at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:216) [!/:3.3.3]
		at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:301) [!/:3.3.3]
		at org.opencastproject.kernel.rest.RestPublisher$RestServlet.handleRequest(RestPublisher.java:558) [!/:?]
		at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doGet(AbstractHTTPServlet.java:225) [!/:3.3.3]
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:687) [!/:3.1.0]
		at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:276) [!/:3.3.3]
		at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:852) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604) [!/:9.4.20.v20190813]
		at org.opencastproject.kernel.rest.JsonpFilter.doFilter(JsonpFilter.java:113) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.opencastproject.kernel.filter.proxy.TransparentProxyFilter.doFilter(TransparentProxyFilter.java:63) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.opencastproject.security.urlsigning.filter.UrlSigningFilter.doFilter(UrlSigningFilter.java:86) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.opencastproject.kernel.security.RemoteUserAndOrganizationFilter.doFilter(RemoteUserAndOrganizationFilter.java:247) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.opencastproject.kernel.rest.CurrentJobFilter.doFilter(CurrentJobFilter.java:87) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:118) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:84) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.opencastproject.kernel.security.AsyncTimeoutRedirectFilter.doFilter(AsyncTimeoutRedirectFilter.java:60) [!/:?]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:113) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:103) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:113) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:146) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:54) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:45) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:98) [!/:3.1.4.RELEASE]
		at org.springframework.security.oauth.provider.filter.OAuthProviderProcessingFilter.doFilter(OAuthProviderProcessingFilter.java:182) [!/:?]
		at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
		at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilter(BasicAuthenticationFilter.java:150) [!/:3.1.7.RELEASE]
		at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
		at org.springframework.security.web.authentication.www.DigestAuthenticationFilter.doFilter(DigestAuthenticationFilter.java:115) [!/:3.1.7.RELEASE]
		at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
		at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:82) [!/:3.1.4.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.authentication.ui.DefaultLoginPageGeneratingFilter.doFilter(DefaultLoginPageGeneratingFilter.java:91) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:183) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:105) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192) [!/:3.1.7.RELEASE]
		at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160) [!/:3.1.7.RELEASE]
		at org.opencastproject.kernel.security.SecurityFilter.doFilter(SecurityFilter.java:124) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.opencastproject.kernel.security.OrganizationFilter.doFilter(OrganizationFilter.java:135) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.opencastproject.kernel.filter.https.HttpsFilter.doFilter(HttpsFilter.java:73) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.opencastproject.kernel.rest.CleanSessionsFilter.doFilter(CleanSessionsFilter.java:89) [!/:?]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1583) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:226) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1591) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:542) [!/:9.4.20.v20190813]
		at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:71) [!/:?]
		at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:536) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1581) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1307) [!/:9.4.20.v20190813]
		at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:293) [!/:?]
		at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:482) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1549) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1204) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [!/:9.4.20.v20190813]
		at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) [!/:?]
		at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.Server.handle(Server.java:494) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:374) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:367) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782) [!/:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918) [!/:9.4.20.v20190813]
		at java.lang.Thread.run(Thread.java:748) [?:1.8.0_262]
```

</details>

`develop` on 2020-08-09

--- a/modules/search/src/main/java/org/opencastproject/matterhorn/search/impl/AbstractElasticsearchIndex.java
+++ b/modules/search/src/main/java/org/opencastproject/matterhorn/search/impl/AbstractElasticsearchIndex.java
@@ -56,8 +56,11 @@ import org.elasticsearch.client.indices.CreateIndexResponse;
 import org.elasticsearch.common.xcontent.XContentType;
 import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.script.Script;
 import org.elasticsearch.search.SearchHits;
 import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.sort.ScriptSortBuilder;
+import org.elasticsearch.search.sort.SortBuilders;
 import org.elasticsearch.search.sort.SortOrder;
 import org.osgi.service.component.ComponentContext;
 import org.osgi.service.component.ComponentException;
@@ -432,12 +435,29 @@ public abstract class AbstractElasticsearchIndex implements SearchIndex {
     // Sort orders
     final Map<String, Order> sortCriteria = query.getSortOrders();
     for (Entry<String, Order> sortCriterion : sortCriteria.entrySet()) {
+      ScriptSortBuilder sortBuilder = null;
+      logger.debug("Event sort criteria: {}", sortCriterion.getKey());
+      if ("publication".equals(sortCriterion.getKey())) {
+        sortBuilder = SortBuilders.scriptSort(
+            new Script("params._source.publication.length"),
+            ScriptSortBuilder.ScriptSortType.NUMBER);
+      }
       switch (sortCriterion.getValue()) {
         case Ascending:
-          searchSource.sort(sortCriterion.getKey(), SortOrder.ASC);
+          if (sortBuilder != null) {
+            sortBuilder.order(SortOrder.ASC);
+            searchSource.sort(sortBuilder);
+          } else {
+            searchSource.sort(sortCriterion.getKey(), SortOrder.ASC);
+          }
           break;
         case Descending:
-          searchSource.sort(sortCriterion.getKey(), SortOrder.DESC);
+          if (sortBuilder != null) {
+            sortBuilder.order(SortOrder.DESC);
+            searchSource.sort(sortBuilder);
+          } else {
+            searchSource.sort(sortCriterion.getKey(), SortOrder.DESC);
+          }
           break;
         default:
           break;

